cardano-cli/src/Cardano/CLI/Byron/Tx.hs

```diff
@@ -211,19 +214,25 @@ issueUTxOExpenditure nc ins outs key =
 -- | Submit a transaction to a node specified by topology info.
 nodeSubmitTx
   :: IOManager
-  -> NodeConfiguration
-  -> SocketPath
+  -> Network
   -> GenTx ByronBlock
-  -> ExceptT RealPBFTError IO ()
-nodeSubmitTx iocp nc targetSocketFp gentx =
-    withRealPBFT nc $
-      \p@Consensus.ProtocolRealPBFT{} -> liftIO $ do
-        -- TODO: Update submitGenTx to use `ExceptT`
-        traceWith stdoutTracer ("TxId: " ++ condense (Consensus.txId gentx))
-        submitGeneralTx iocp targetSocketFp
-                        (pInfoConfig (Consensus.protocolInfo p))
-                        gentx
-                        nullTracer -- stdoutTracer
+  -> ExceptT CliError IO ()
+nodeSubmitTx iomgr network gentx = do
+    socketPath <- readEnvSocketPath
+    liftIO $ do
+      traceWith stdoutTracer ("TxId: " ++ condense (Consensus.txId gentx))
+      _res <- submitGenTx
+                nullTracer -- stdoutTracer
+                iomgr
+                (protocolClientInfo ptcl)
+                network
+                socketPath
+                gentx
+      --TODO: print failures
```

Comment:
Or rather, return them as a CLiError.

Reply:
Yes, this will have to be separated into two functions, one lib function (used in this module and for votes, update proposals etc) and another one that is just the cli command impl, which needs to print the failure on the console for the user.